mesonbuild/modules/gnome.py

```diff
@@ -408,13 +408,21 @@ def fix_ldflags(ldflags):
         else:
             return cflags, internal_ldflags, external_ldflags, external_ldflags_nodedup, gi_includes
 
-    def _unwrap_gir_target(self, girtarget):
+    def _unwrap_gir_target(self, girtarget, state):
         while hasattr(girtarget, 'held_object'):
             girtarget = girtarget.held_object
+
         if not isinstance(girtarget, (build.Executable, build.SharedLibrary,
                                       build.StaticLibrary)):
             raise MesonException('Gir target must be an executable or library')
 
+        STATIC_BUILD_REQUIRED_VERSION = ">=1.58.1"
+        if isinstance(girtarget, (build.StaticLibrary)) and \
+           not mesonlib.version_compare(
+               self._get_gir_dep(state)[0].get_version(),
```

Comment:
There is already a call to _get_gir_dep() in generate_gir(). You could just move that call up, so you can use self.git_dep here.

Reply:
_get_gir_dep() seems to just reutnr self.gir_dep if it already exists, seems a little less brittle than depending on the ordering.